File: ISL.ReIdentification.WebApps.Configurations/ISL.ReIdentification.Configurations.Client/src/models/impersonationContext/impersonationContext.ts
export class ImpersonationContext {
    public id: string;
    public requesterFirstName: string;
    public requesterLastName: string;
    public requesterEmail: string;
    public recipientFirstName: string;
    public recipientLastName: string;
    public recipientEmail: string;
    public reason: string;
    public purpose: string;
    public organisation: string;
    public isApproved: boolean;
    public identifierColumn: string;
    public createdBy?: string;
    public createdDate?: Date | undefined;
    public updatedBy?: string;
    public updatedDate?: Date | undefined;

    constructor(impersonationContext: ImpersonationContext) {
        this.id = impersonationContext.id ? impersonationContext.id : "";
        this.requesterFirstName = impersonationContext.requesterFirstName || "";
        this.requesterLastName = impersonationContext.requesterLastName || "";
        this.requesterEmail = impersonationContext.requesterEmail || "";
        this.recipientFirstName = impersonationContext.recipientFirstName || "";
        this.recipientLastName = impersonationContext.recipientLastName || "";
        this.recipientEmail = impersonationContext.recipientEmail || "";
        this.reason = impersonationContext.reason || "";
        this.purpose = impersonationContext.purpose || "";
        this.organisation = impersonationContext.organisation || "";
        this.isApproved = impersonationContext.isApproved || false;
        this.identifierColumn = impersonationContext.identifierColumn || ""; 
        this.createdDate = impersonationContext.createdDate ? new Date(impersonationContext.createdDate) : undefined;
        this.createdBy = impersonationContext.createdBy || "";
        this.updatedDate = impersonationContext.updatedDate ? new Date(impersonationContext.updatedDate) : undefined;
        this.updatedBy = impersonationContext.updatedBy || "";
    }
}

File: ISL.ReIdentification.Core/Models/Foundations/ImpersonationContexts/ImpersonationContext.cs
// ---------------------------------------------------------
// Copyright (c) North East London ICB. All rights reserved.
// ---------------------------------------------------------

using System;

namespace ISL.ReIdentification.Core.Models.Foundations.ImpersonationContexts
{
    public class ImpersonationContext : IKey, IAudit
    {
        public Guid Id { get; set; }
        public Guid RequesterEntraUserId { get; set; } = Guid.Empty;
        public string RequesterFirstName { get; set; } = string.Empty;
        public string RequesterLastName { get; set; } = string.Empty;
        public string RequesterDisplayName { get; set; } = string.Empty;
        public string RequesterEmail { get; set; } = string.Empty;
        public string RequesterJobTitle { get; set; } = string.Empty;
        public Guid RecipientEntraUserId { get; set; } = Guid.Empty;
        public string RecipientFirstName { get; set; } = string.Empty;
        public string RecipientLastName { get; set; } = string.Empty;
        public string RecipientDisplayName { get; set; } = string.Empty;
        public string RecipientEmail { get; set; } = string.Empty;
        public string RecipientJobTitle { get; set; } = string.Empty;
        public string Reason { get; set; } = string.Empty;
        public string Purpose { get; set; } = string.Empty;
        public string Organisation { get; set; } = string.Empty;
        public string PipelineName { get; set; } = string.Empty;
        public string ManagedIdentityName { get; set; } = string.Empty;
        public bool IsApproved { get; set; }
        public string IdentifierColumn { get; set; } = string.Empty;
        public string CreatedBy { get; set; } = string.Empty;
        public DateTimeOffset CreatedDate { get; set; }
        public string UpdatedBy { get; set; } = string.Empty;
        public DateTimeOffset UpdatedDate { get; set; }
    }
}